modules/mono/glue/cs_files/Mathf.cs

```diff
@@ -181,6 +238,57 @@ public static real_t Min(real_t a, real_t b)
             return (a < b) ? a : b;
         }
 
+        public static real_t Mod(real_t a, real_t b)
+        {
+            real_t c = a % b;
+            if (c < 0) 
+            {
+                c += b;
+            }
+            return c;
+        }
+
+        public static int ModInt(real_t a, real_t b)
+        {
+            real_t c = a % b;
+            if (c < 0) 
+            {
+                c += b;
+            }
+            return RoundToInt(c);
+        }
+
+        public static int ModInt(int a, int b)
+        {
+            int c = a % b;
+            if (c < 0) 
+            {
+                c += b;
+            }
+            return c;
+        }
+
+        /// <summary>
+        /// Very fast Modulus function using a bitwise operation. 
+        /// B is only valid for powers of two, such as 2, 4, 8, 16, etc.
+        /// If you are looking for pure speed, see if ModBit2 works for you.
+        /// </summary>
+        public static int ModPow2(int a, int b)
+        {
+            return (a & (b-1));
+        }
+
+        /// <summary>
+        /// Extremely fast Modulus function using a bitwise operation. 
+        /// B is only valid for powers of two minus one, such as 1, 3, 7, 15, etc.
+        /// Output is the same as ModPow2 with the corresponding power of two.
+        /// For example, `ModBit2(a, 15)` is the same as `ModPow2(a, 16)`.
+        /// </summary>
+        public static int ModBit2(int a, int b)
+        {
+            return (a & b);
```

Comment:
Its a basic operation between two integers, the user can do it himself, without calling a function(which is not a fast operation)..

Reply:
AFAIK C# JIT has an inline optimization for methods that are compiled to 32 bytes or less.

`ModBit2` does the same as `&`, yes, but it states the purpose of the operation in code. I also have this method defined in Point classes so it's nice to have a method in Mathf.

Reply:
Yes, the compiler does a good job at optimizing this. In fact, I was testing the other day with a conversion operator between two Vector3 types:

``` C#
public static implicit operator Godot.Vector3(OpenTK.Vector3 from)
{
    return new Godot.Vector3(from.X, from.Y, from.Z);
}


OpenTK.Vector3 a = new OpenTK.Vector3();
Godot.Vector3 b = a;
```

I was very surprise to see that the compiler optimizes it to be the same as a fast memory copy:

``` C#
Godot.Vector3 a = new Godot.Vector3();
Godot.Vector3 b = a;
```